llvm/include/llvm/CodeGen/SelectionDAG.h

```diff
@@ -2111,6 +2115,10 @@ class SelectionDAG {
   /// Test whether the given SDValue is known to contain non-zero value(s).
   bool isKnownNeverZero(SDValue Op, unsigned Depth = 0) const;
 
+  /// Test whether the given float value is known to be positive. +0.0, +inf and
+  /// +nan are considered positive, -0.0, -inf and -nan are not.
+  bool isKnownPositiveOrNaNFP(SDValue Op) const;
```

Comment:
Best to follow the ValueTracking naming and call this cannotBeOrderedLessThanZero

Reply:
Right, you can only call it that if you're happy with the semantics change too, which is that it would return true for -0.0.

Reply:
> bitwise negative

Not quite, sign bit of nan is meaningless. How about cannotBeOrderedNegativeFP?

Reply:
I see, The IEEE 754 standard specifies that the sign bit of a NaN has no significance. The Arm architecture is obviously more opinionated about what the correct answer is, and I have been testing bitwise.

That probably helps with copysign too. I can change it to the new name.